### PR TITLE
feat(utxo-lib): add integration test for p2trMusig2

### PR DESCRIPTION
**Pull request by saravanan7mani**
Add deposit and spend integration test for key path and script path `p2trMusig2`.
This PR tests PSBT build, nonce creation, sign, verify and finalizer works against a `regtest` node.
Testing p2trMusig2 with parse test suite is not covered in this PR.
p2trMusig2 support for `parseSignatureScript`, `UtxoPsbt.fromTransaction` are required here.
And now those are not that critical, so will revisit parse test case later when required.

TICKET: BG-71051

### CODE REVIEW

File: modules/utxo-lib/test/integration_local_rpc/generate/main.ts [hunk 1]
```diff
@@ -142,24 +156,48 @@ async function createTransactionsForScriptType(
...
+      } else {
+        spendTx = createSpendTransaction(
+          keys,
+          scriptType,
+          [deposit1Tx, deposit2Tx],
+          script,
+          protocol.network,
+          protocol.version
+        );
+      }
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/test/integration_local_rpc/generate/main.ts:185]: let's move this to a `getSpendTx` func

File: modules/utxo-lib/test/integration_local_rpc/generate/outputScripts.util.ts [hunk 2]
```diff
@@ -158,3 +168,92 @@ export function createSpendTransaction<TNumber extends number | bigint = number>
...
+}
+
+/**
+ * @returns BIP32 hardcoded index for p2trMusig2 spend type. 0 for key path and 100 for script path.
+ * For same fixture key triple and script type (p2trMusig2),
+ * we need 2 different deposit and spend tx fixtures.
+ */
+export function getP2trMusig2Index(spendType: 'KeyPath' | 'ScriptPath'): number {
+  return spendType === 'KeyPath' ? 0 : 100;
+}
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/test/integration_local_rpc/generate/outputScripts.util.ts:259]: nit: we tend to use camelcase with lowercase initial character in utxolib

```suggestion

export type Musig2SpendType = 'keyPath' | 'scriptPath`;

/**
 * @returns BIP32 hardcoded index for p2trMusig2 spend type. 0 for key path and 100 for script path.
 * For same fixture key triple and script type (p2trMusig2),
 * we need 2 different deposit and spend tx fixtures.
 */
export function getP2trMusig2Index(spendType: SpendPath): number {
  return spendType === 'keyPath' ? 0 : 100;
}
```

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
two nits

File: modules/utxo-lib/test/integration_local_rpc/generate/outputScripts.util.ts (hunk 2 above)
Comment (saravanan7mani) [reply modules/utxo-lib/test/integration_local_rpc/generate/outputScripts.util.ts]: Done

File: modules/utxo-lib/test/integration_local_rpc/generate/main.ts (hunk 1 above)
Comment (saravanan7mani) [reply modules/utxo-lib/test/integration_local_rpc/generate/main.ts]: Done

**Review by OttoAllmendinger — APPROVED** [review]

**Pull request merged**